rust/lance/src/index/vector.rs

```diff
@@ -253,57 +254,39 @@ pub(crate) async fn build_vector_index(
-            .ok_or(Error::Schema {
-                message: format!("Column {} not found in schema", column),
-                location: location!(),
-            })?
-            .data_type();
-        match data_type {
-            DataType::FixedSizeList(f, _) => match f.data_type() {
-                DataType::Float16 | DataType::Float32 | DataType::Float64 => {
-                    IvfIndexBuilder::<FlatIndex, FlatQuantizer>::new(
-                        dataset.clone(),
-                        column.to_owned(),
-                        dataset.indices_dir().child(uuid),
-                        params.metric_type,
-                        Box::new(shuffler),
-                        Some(ivf_params.clone()),
-                        Some(()),
-                        (),
-                    )?
-                    .build()
-                    .await?;
-                }
-                DataType::UInt8 => {
-                    IvfIndexBuilder::<FlatIndex, FlatBinQuantizer>::new(
-                        dataset.clone(),
-                        column.to_owned(),
-                        dataset.indices_dir().child(uuid),
-                        params.metric_type,
-                        Box::new(shuffler),
-                        Some(ivf_params.clone()),
-                        Some(()),
-                        (),
-                    )?
-                    .build()
-                    .await?;
-                }
-                _ => {
-                    return Err(Error::Index {
-                        message: format!(
-                            "Build Vector Index: invalid data type: {:?}",
-                            f.data_type()
-                        ),
-                        location: location!(),
-                    });
-                }
-            },
+        let element_type = get_vector_element_type(dataset, column)?;
+        match element_type {
+            DataType::Float16 | DataType::Float32 | DataType::Float64 => {
+                IvfIndexBuilder::<FlatIndex, FlatQuantizer>::new(
+                    dataset.clone(),
+                    column.to_owned(),
+                    dataset.indices_dir().child(uuid),
+                    params.metric_type,
+                    Box::new(shuffler),
+                    Some(ivf_params.clone()),
+                    Some(()),
+                    (),
+                )?
+                .build()
+                .await?;
+            }
+            DataType::UInt8 => {
+                IvfIndexBuilder::<FlatIndex, FlatBinQuantizer>::new(
+                    dataset.clone(),
+                    column.to_owned(),
+                    dataset.indices_dir().child(uuid),
+                    params.metric_type,
+                    Box::new(shuffler),
+                    Some(ivf_params.clone()),
+                    Some(()),
+                    (),
+                )?
+                .build()
+                .await?;
+            }
```

Comment:
Why did this change?

Reply:
I noticed there are many lines are doing the same thing: get the vector data type / value type and check it.
so just made the function `get_vector_element_type` to do this